chalk-solve/src/clauses/builtin_traits/unsize.rs

```diff
@@ -0,0 +1,498 @@
+use std::collections::HashSet;
+use std::iter;
+
+use crate::clauses::ClauseBuilder;
+use crate::{Interner, RustIrDatabase, TraitRef, WellKnownTrait};
+use chalk_ir::{
+    cast::Cast,
+    interner::HasInterner,
+    visit::{visitors::FindAny, SuperVisit, Visit, VisitResult, Visitor},
+    ApplicationTy, Binders, Const, ConstValue, DebruijnIndex, DomainGoal, DynTy, EqGoal, Goal,
+    LifetimeOutlives, QuantifiedWhereClauses, Substitution, TraitId, Ty, TyData, TypeName,
+    WhereClause,
+};
+
+struct UnsizeParameterCollector<'a, I: Interner> {
```

Comment:
Ooof but this file needs some comments. =) But I don't think it has to block landing necessarily. But we should file an issue  or something to follow-up and explain what is going on.